src/main.c

```diff
@@ -1,51 +1,51 @@
 #include <stdio.h>
 #include <stdint.h>
+
 #include "lcd.h"
 #include "cpu.h"
 #include "rom.h"
 #include "mmu.h"
+#include "errcodes.h"
 
 int main(void) {
-    uint8_t ret = rom_load("./roms/hello-world.gb");
+    ErrorCode ret = rom_load("./roms/hello-world.gb");
     if (ret) {
         fprintf(stderr, "Failed to load ROM\n");
-        exit(-1);
+        exit((int)ret);
```

Comment:
Why are you casting the ret value? enums (usually) are ints, right? 🤔

Reply:
That's what I thought, too. Apparently it is not, as some warning comes up if compiling without the cast.

Reply:
That's interesting. I think that the compiler is free to pick a type as long as it's integer, so yeah, could he unsigned or something :'D
But not much of an issue anyway

Reply:
I agree that this should not stay the way it is.
Let's investigate this issue further because this casting could get quite tedious down the line

Reply:
If building with clang, building fails due to implicit casting here.

Reply:
Considering your first comment for src/main.c:14: This just has to be used in main(). No casting is necessary anywhere else, if ErrorCodes gets used consistently.